Expo1900/Expo1900/Utils/DecodingHandler.swift

```diff
@@ -0,0 +1,40 @@
+//
+//  DecodingHandler.swift
+//  Expo1900
+//
+//  Created by Minsup, Jusbug on 2023/06/28.
+//
+
+import Foundation
+
+enum DecodingHandler {
+    enum FileName: String {
+        case exposition = "exposition"
+        case entry = "entry"
+        
+        var name: String {
+            return self.rawValue
+        }
+    }
+    
+    static func decode<T: Decodable>(fileName file: FileName) -> T? {
```

Comment:
아래와 같이 제네릭에 Decodable 자체를 넣으면 어떻게 될까요? 굳이 `T: Decodable`로 선언한 이유가 있을까요?
```suggestion
    static func decode<Decodable>(fileName file: FileName) -> Decodable? {
```

Reply:
문법 오류가 발생합니다. 
`<T: Decodable>`의 뜻은 Decodable을 준수한 타입을 사용하겠다는 뜻이고 T는 함수 내부와 리턴에서 사용해야하기 때문에 T와 같이 사용할 타입 이름을 명시해 주어야합니다.

Reply:
> T는 함수 내부와 리턴에서 사용해야하기 때문에 T와 같이 사용할 타입 이름을 명시해 주어야합니다.

이 부분을 조금 더 자세하게 설명해주시겠어요? 명시를 하지 않는다면 컴파일러는 어떻게 받아들일까요? 그리고 그게 왜 문제가 될까요?

Reply:
* Generic은 `어떤 타입`에 대해서도 대응할 수 있는 메소드나 타입을 만들기 위해 존재합니다.
* 여기서 `어떤 타입`이라는 것은 특정 타입이 아닌, 매우 광범위하고 추상적인 타입을 의미하기 때문에 특별한 별칭이 필요합니다.
* 만약 별칭을 지어주지 않는다면 `어떤 타입`을 언급할 방법이 없기 때문입니다.
* 사람도 이름이 없으면 부를 수 없는 것 처럼 `어떤 타입`도 이름을 붙여야 원하는 시기와 장소에서 사용할 수 있게 됩니다.

위에 예시처럼

```swift
static func decode<Decodable>(fileName file: FileName) -> Decodable? {
```
이런 코드를 작성한다고 가정했을 때 
별칭을 정해주지 않았기 때문에 함수 내부에서 `어떤 타입`을 사용할 수 없게 됩니다.

```swift
guard let result = try? decoder.decode( ??? .self, from: content) else {
    print("\(file).json 파일을 \( ??? .self)으로 변환하는데 실패하였습니다.") 
    return nil
}
```